### PR TITLE
Dna predict language modeling fixes for getting predict task work for dna datasets

### PR DESCRIPTION
**Pull request by deysanjoy33**

### CODE REVIEW

File: run/dna_predict.yaml [hunk 1]
```diff
@@ -30,5 +43,6 @@ checkpoint: null
 
 batch_size: 32
 dataset_name: coreprom
-#label_column_name: label
-input_directory: /proj/bmfm/datasets/omics/genome/finetune_datasets/core_promoter_prediction
+label_column_name: label
+input_directory: /proj/bmfm/datasets/omics/genome/finetune_datasets/core_promoter_prediction/test.csv
```
[the remark this answers is not in the document]
Comment (deysanjoy33) [reply run/dna_predict.yaml:47]: It works for both directory or file input in dataset_kwargs.processed_data_source. If input_directory is provided with train/test/dev, then embedding is done on combined dataset, o/w a specific dataset can be provided as input_file arg.

File: bmfm_targets/training/data_module.py [hunk 2]
```diff
@@ -877,6 +877,15 @@ def __init__(
...
         sequence_dropout_factor: int | float | None = None,
+        balancing_label_column: str | None = None,
+        mlm: bool = False,
+        change_ratio: float = 0.15,
+        mask_ratio: float = 1.0,
+        switch_ratio: float = 0.0,
+        masking_strategy: MaskingStrategy | None = None,
+        prevent_attention_to_masked: bool = False,
+        comask_across_fields: bool = False,
+        tokenize_kwargs: dict | None = None,
```
[the remark this answers is not in the document]
Comment (deysanjoy33) [reply bmfm_targets/training/data_module.py:888]: Bharat wanted to make it separate intentionally, since many of the RNA transforms, special data formats are not needed in DNA datasets.

File: run/dna_predict.yaml [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
 
 task:
-  checkpoint: ${checkpoint}
   output_embeddings: true
   output_predictions: true
   default_root_dir: ${working_dir}
 
+
+trainer:
+  batch_size: ${batch_size}
```
[the remark this answers is not in the document]
Comment (deysanjoy33) [reply run/dna_predict.yaml:30]: May be a dumb question, if the batch size in trainer is redundant, then why is it not removed from TrainerConfig and only data_module.batch_size is used all through?  I see that although it is passed in task_utils.instantiate_module_from_checkpoint(), it is not used in the base.py training module.

**Review by mmdanziger — CHANGES_REQUESTED** [review]
my concern with copying over code into the dna datamodule is:

1. code repetition tends to invite subtle behavior differences and bugs over time
2. some of these “features” aren’t actually supported — they don’t work (eg wced) or have never been tested — so it can give the impression this dm does things it can’t really do
3. i’m not sure this is the right solution to the actual problem, which i understand is running inference. why can inference not work with the existing dm? what’s the real blocker?

File: run/dna_predict.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (deysanjoy33) [reply run/dna_predict.yaml:47]: Ok.

File: bmfm_targets/training/data_module.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (deysanjoy33) [reply bmfm_targets/training/data_module.py:888]: No, this DM was mainly used only for finetuning so far. But, to make the predict() function work with 'language_modeling' option, (because the other options such as 'sequence_classification' or 'multitask' did not work without proper loss function, etc. defined) I had to use the masking defined here at least in init() to be consistant with the task_utils functions.

**Review by mmdanziger — APPROVED** [review]
Thank you @deysanjoy33 ! Big boost for BMFM-DNA usability :)
